packages/block/src/header.ts

```diff
@@ -67,89 +162,44 @@ export class BlockHeader {
         this._common = new Common({ chain: DEFAULT_CHAIN })
       }
     }
-    const fields = [
-      {
-        name: 'parentHash',
-        length: 32,
-        default: zeros(32),
-      },
-      {
-        name: 'uncleHash',
-        default: KECCAK256_RLP_ARRAY,
-      },
-      {
-        name: 'coinbase',
-        length: 20,
-        default: zeros(20),
-      },
-      {
-        name: 'stateRoot',
-        length: 32,
-        default: zeros(32),
-      },
-      {
-        name: 'transactionsTrie',
-        length: 32,
-        default: KECCAK256_RLP,
-      },
-      {
-        name: 'receiptTrie',
-        length: 32,
-        default: KECCAK256_RLP,
-      },
-      {
-        name: 'bloom',
-        default: zeros(256),
-      },
-      {
-        name: 'difficulty',
-        default: Buffer.from([]),
-      },
-      {
-        name: 'number',
-        // TODO: params.homeSteadForkNumber.v left for legacy reasons, replace on future release
-        default: toBuffer(1150000),
-      },
-      {
-        name: 'gasLimit',
-        default: Buffer.from('ffffffffffffff', 'hex'),
-      },
-      {
-        name: 'gasUsed',
-        empty: true,
-        default: Buffer.from([]),
-      },
-      {
-        name: 'timestamp',
-        default: Buffer.from([]),
-      },
-      {
-        name: 'extraData',
-        allowZero: true,
-        empty: true,
-        default: Buffer.from([]),
-      },
-      {
-        name: 'mixHash',
-        default: zeros(32),
-        // length: 32
-      },
-      {
-        name: 'nonce',
-        default: zeros(8), // sha3(42)
-      },
-    ]
-    defineProperties(this, fields, data)
+
+    this.parentHash = parentHash
+    this.uncleHash = uncleHash
+    this.coinbase = coinbase
+    this.stateRoot = stateRoot
+    this.transactionsTrie = transactionsTrie
+    this.receiptTrie = receiptTrie
+    this.bloom = bloom
+    this.difficulty = difficulty
+    this.number = number
+    this.gasLimit = gasLimit
+    this.gasUsed = gasUsed
+    this.timestamp = timestamp
+    this.extraData = extraData
+    this.mixHash = mixHash
+    this.nonce = nonce
 
     if (options.hardforkByBlockNumber) {
-      this._common.setHardforkByBlockNumber(bufferToInt(this.number))
+      this._common.setHardforkByBlockNumber(this.number.toNumber())
```

Comment:
We should make common support BNs as input here.

Reply:
@holgerd77 do you think we should include this (and [this](https://github.com/ethereumjs/ethereumjs-vm/pull/883#discussion_r496244200)) in this PR or open as a separate one?

Reply:
Can be included here I would say.

Reply:
I started to add optional param type `BN` for `setHardforkByBlockNumber` and `activeHardfork` but unless we use BNs internally there is not much benefit here, and just adds code `if (BN.isBN(blockNumber) { blockNumber = blockNumber.toNumber() }` in a handful of places.

I think it is more simple to just convert the BN `toNumber()` when calling the function as you did here. If we do think blockNumbers will exceeds 53 bits in the future, then we should probably take on this work in a separate PR as a different effort.

So I will resolve this for now, please chime in if you feel strongly otherwise.

Reply:
Yes, think that makes sense.